demos/word_count_demo.cc

```diff
@@ -0,0 +1,372 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2023 ScyllaDB Ltd.
+ */
+
+#include <iostream>
+#include <memory>
+#include <seastar/core/aligned_buffer.hh>
+#include <seastar/core/app-template.hh>
+#include <seastar/core/fstream.hh>
+#include <seastar/core/seastar.hh>
+#include <seastar/core/when_all.hh>
+#include <seastar/util/log.hh>
+
+namespace {
+seastar::logger logger("word_count");
+
+// The page size for doing I/O operations.
+static constexpr size_t kPageSize = 4096;
+
+/**
+ * An abstract class that represent the task to be performed on the file. Each
+ * task is associated with the FilePrcessor.
+ */
+class FileProcessorTask {
```

Comment:
Seastar coding style suggest camel-case for template parameters _only_

Reply:
Thank you for pointing this out. I have aligned the code according to the guidelines.